File: src/main/node/magma/list/SetIter.ts
import Iterator from "../../java/util/Iterator";
import Set from "../../java/util/Set";
/** Iterator backed by a java.util.Set. */
export default class SetIter<T> implements Iter<T> {
    private readonly iterator: Iterator<T>;

    SetIter(iterator: Iterator<T>): private {
        // TODO
    }

    /** Wrap an existing Set. */
    wrap(set: Set<T>): SetIter<T> {
        return new SetIter<T>(set.iterator());
    }

    @Override
    hasNext(): boolean {
        return iterator.hasNext();
    }

    @Override
    next(): T {
        return iterator.next();
    }
}

File: src/main/java/magma/list/SetIter.java
package magma.list;

import java.util.Iterator;
import java.util.Set;

/** Iterator backed by a java.util.Set. */
public class SetIter<T> implements Iter<T> {
    private final Iterator<T> iterator;

    private SetIter(Iterator<T> iterator) {
        this.iterator = iterator;
    }

    /** Wrap an existing Set. */
    public static <T> SetIter<T> wrap(Set<T> set) {
        return new SetIter<>(set.iterator());
    }

    @Override
    public boolean hasNext() {
        return iterator.hasNext();
    }

    @Override
    public T next() {
        return iterator.next();
    }
}
